lib/proj-parser.js

```diff
@@ -0,0 +1,77 @@
+var fs = require('fs');
+var path = require('path');
+var parseXML = require('xml2js').parseString;
+var debug = require('debug')('snyk');
+
+function determineDotnetVersion(rootDir) {
+  var csprojPath = findFile(rootDir, /.*\.csproj$/);
+  if (!csprojPath) {
+    throw new Error('.csproj file not found in ' + rootDir + '. ' +
+      'Make sure project was built before running `snyk test`');
+  }
+  debug('Checking .net framework version in .csproj file ' +
+    csprojPath);
+
+  var csprojContents = fs.readFileSync(csprojPath);
+
+  var frameworks = [];
+  parseXML(csprojContents, function (err, parsedCsprojContents) {
+    if (err) {
+      throw err;
+    }
+
+    var versionLoc = parsedCsprojContents.Project.PropertyGroup[0];
```

Comment:
what if parsedCsprojContents.Project.PropertyGroup[0] / parsedCsprojContents.Project / parsedCsprojContents.Project.PropertyGroup doesn't exist?

Reply:
parsedCsprojContents.Project has to exist in the csproj, or else it is not a valid csproj. 
Added a check for PropertyGroup because looks like that is an optional field.

Reply:
add lodash